### PR TITLE
*: add docker support.

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @qiuyesuifeng

### CODE REVIEW

**Comment by qiuyesuifeng** [comment]
LGTM

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM pingcap/rust
+
+MAINTAINER siddontang
+
+ADD . /tikv
+
+RUN cd /tikv && \
+    cargo build --release && \
+    cp -f target/release/tikv-server /tikv-server && \
```
Comment (BusyJay) [inline Dockerfile]: Maybe use `/bin/tikv-server`?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    rm -rf /tikv
+
+EXPOSE 20160
+
+ENTRYPOINT ["/tikv-server"]
```
Comment (BusyJay) [inline Dockerfile]: new line.

File: docker/rust_dockerfile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    libgflags-dev \
+    libsnappy-dev \
+    zlib1g-dev \
+    libbz2-dev \
+    git \
+    ca-certificates \
+    && rm -rf /var/lib/apt/lists/* && apt-get clean
+
+RUN cd / && \
+    git clone --depth=1 https://github.com/facebook/rocksdb.git && \
```
Comment (BusyJay) [inline docker/rust_dockerfile]: I think use curl to download archive maybe better.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+MAINTAINER siddontang
+
+ADD . /tikv
+
+RUN cd /tikv && \
+    cargo build --release && \
+    cp -f target/release/tikv-server /tikv-server && \
+    rm -rf /tikv
+
+EXPOSE 20160
```
Comment (BusyJay) [inline Dockerfile]: Is 20160 hard coded in tikv-server?

File: Dockerfile (hunk 1 above)
Comment (siddontang) [inline Dockerfile]: no need, many docker use root "/" as the entrypoint.

File: Dockerfile (hunk 4 above)
Comment (siddontang) [inline Dockerfile]: yes

**Comment by ngaut** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @BusyJay

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**